clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2655,6 +2960,87 @@ class SyclKernelBodyCreator : public SyclKernelFieldHandler {
     addFieldInit(FD, Ty, ParamRef);
   }
 
+  Expr *addDerivedToBaseCastExpr(const CXXRecordDecl *RD,
+                                 const CXXBaseSpecifier &BS,
+                                 Expr *LocalCloneRef) {
+    CXXCastPath BasePath;
+    QualType DerivedTy(RD->getTypeForDecl(), 0);
+    QualType BaseTy = BS.getType();
+    SemaRef.CheckDerivedToBaseConversion(DerivedTy, BaseTy, KernelCallerSrcLoc,
+                                         SourceRange(), &BasePath,
+                                         /*IgnoreBaseAccess*/ true);
+    auto Cast = ImplicitCastExpr::Create(
+        SemaRef.Context, SemaRef.Context.getPointerType(BaseTy),
+        CK_DerivedToBase, LocalCloneRef,
+        /* CXXCastPath=*/&BasePath, VK_LValue, FPOptionsOverride());
+    return Cast;
+  }
+
+  Expr *getAddressOf(Expr *E) {
```

Comment:
```suggestion
  Expr *createGetAddressOf(Expr *E) {
```
It doesn't get address, right? It creates an operator that gets address.